nRepl server start stop and API (#29)
BetterThanTomorrow/joyride: doc/api.md

# APIs

There are two different APIs here:

1. The Joyride *Scripting API*
1. The Joyride *Extension API*

> Please note that Joyride's *Extension API* is available from *Joyride scripts*.

Reviewer: Does you mean, _also_ available?

Author: ```suggestion
Please note that Joyride's *Extension API* is also available to *Joyride scripts*.
```

## Possibly coming additions

We want to add `clojure.test` and `clojure.pprint` as well in the near future. How near/if depends on things like how much time we can spend on it, and how easy/hard it will be to port this over from [nbb](https://github.com/babashka/nbb).

## Extension API

Joyride's Extension API has two parts:

1. [`when` clauses contexts](https://code.visualstudio.com/api/references/when-clause-contexts)
1. The `exports` map/object on the activated [extension](https://code.visualstudio.com/api/references/vscode-api#extensions) instance.

### `when` clause context

The following contexts are available for users of Joyride when binding commands to keyboard shortcuts:

> * `joyride.isActive`, `boolean` - Wether the joyRide extension is active or not

Reviewer: spelling: whether

> * `joyride.isActive`, `boolean` - Wether the joyRide extension is active or not

Author: ```suggestion
* `joyride.isActive`, `boolean` - Whether the joyRide extension is active or not
```

> * `joyride.isNReplServerRunning`, `boolean` - Wether the Joyride nREPL server is running or not

Reviewer: Same

Author: ```suggestion
* `joyride.isNReplServerRunning`, `boolean` - Whether the Joyride nREPL server is running or not
```
